### PR TITLE
BFD-1403: Update patient discharge status code mapping

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-1403](https://jira.cms.gov/browse/BFD-1403)

**User Story or Bug Summary:**

Patient discharge status code correctly mapped in supporting info.
---

### What Does This PR Do?

Modification of the transformerUtils in V2 to map patient discharge status code to patientDischargeStatusCode (db field) instead of claimFrequencyCode
- changed V2 transformerUtils to update patientDischargeStatusCode mapping
- modified hha and inpatient unit tests to account for updated mapping 
- updated json files with new status and display codes for hha, hospice, inpatient for patientDischargeStatusCode mapping

### What Should Reviewers Watch For?

Does the updated mapping in transformerUtils look correct and avoid any issues for end users if shipped?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by christrotteradhoc — CHANGES_REQUESTED** [review]
Looks good! I'll approve once we have the comms in place.

**Review by mackec1 — APPROVED** [review]
LGTM.

File: apps/bfd-server/dev/api-changelog.md [hunk 1]
```diff
@@ -1,5 +1,44 @@
 # API Changelog
 
+## BFD-1403 Update patient discharge status code mapping
+
+Update mapping for patient discharge status coding for HHA, Inpatient and Hospice:
```
Comment (ajsarkar128) [inline apps/bfd-server/dev/api-changelog.md:5]: Change to "Update mapping for patient discharge status coding for HHA, Inpatient and Hospice. Previously, the patient discharge status code in the EOB FHIR response was incorrectly being populated by the claim frequency code data field. The value for the patient discharge status code is now correctly being populated by the patient discharge status code data field."

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java [hunk 2]
```diff
@@ -274,7 +274,9 @@ public void shouldHaveDischargeStatusSupInfo() {
                     "Discharge Status")),
             // Code
             new Coding(
-                "https://bluebutton.cms.gov/resources/variables/ptnt_dschrg_stus_cd", "1", null));
+                "https://bluebutton.cms.gov/resources/variables/ptnt_dschrg_stus_cd",
+                "30",
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java:278]: I'm seeing how these mapping errors occur; we're just using a hardcoded value that isn't pulled from the source data field in the tests. Nothing to do for this PR, but it seems like these tests could use a better expected value by pulling from the expected mapping location of the input, so we make sure ccw data input -> output path/value is correct.

**Review by lsmitchell — DISMISSED** [review]

**Review by christrotteradhoc — DISMISSED** [review]

**Review by lsmitchell — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**